test/tasks/appium.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+var spawn = require('child_process').spawn,
+    defaultOptions = {
+        args: []
+    };
+
+module.exports = function (grunt) {
+    var appium;
+    function kill(err) {
+        if (err)
+            console.error(err);
+        if (appium)
+            appium.kill('SIGINT');
+    }
+
+    process.on('exit', kill);
+    process.on('SIGINT', kill);
+    process.on('uncaughtException', kill);
```

Comment:
How about moving these event listener into the appium task? If they are Appium specific they should rather live there otherwise in the main grunt file